pkg/storage/chunk/client/util/reader.go

```diff
@@ -0,0 +1,17 @@
+package util
+
+import (
+	"bytes"
+	"io"
+)
+
+func ReadSeeker(r io.Reader) (io.ReadSeeker, error) {
+	if rs, ok := r.(io.ReadSeeker); ok {
+		return rs, nil
+	}
+	data, err := io.ReadAll(r)
```

Comment:
Should we think about a sync.Pool here? Presumably it allocates a new buffer every time
Should be fine for now, we can return to it if it's an issue.

Reply:
I think the caller should really pass a ReadSeeker, this is a best effort.